src/store/modules/navigation.ts

```diff
@@ -109,13 +109,30 @@ const mutations = {
 
   setActiveFoodCategory(state: NavState, newCategory: string) {
     state.activeFoodCategory = newCategory;
+  },
+
+  filterFoodByCategory(state: NavState){
+    if(state.activeFoodCategory === "all"){
```

Comment:
try avoiding comparing stuff to string values. it would be much better when category name would be some constant that you can import from any place in code. then, when it changes for some reason, you wont have to search in entire codebase to change all entieties.